adobe_vipm/flows/sync.py

```diff
@@ -1126,6 +1142,51 @@ def sync_agreement(  # noqa: C901
             _update_last_sync_date(mpt_client, agreement)
 
 
+def _process_main_agreement(
+        mpt_client: MPTClient,
+        adobe_client: AdobeClient,
+        agreement: dict,
+        customer: dict,
+        adobe_deployments: list[dict],
+        adobe_subscriptions: list[dict],
+        *,
+        dry_run: bool,
+        sync_prices: bool
+):
+    _check_update_airtable_missing_deployments(
+        agreement, adobe_deployments, adobe_subscriptions, dry_run=dry_run
+    )
+    deployment_agreements = get_agreements_by_customer_deployments(
+        mpt_client,
+        Param.DEPLOYMENT_ID.value,
+        [deployment["deploymentId"] for deployment in adobe_deployments],
+    )
+    active_deployment_agreements = [
+        agreement
+        for agreement in deployment_agreements
+        if agreement["status"] == AgreementStatus.ACTIVE
+    ]
+    if active_deployment_agreements:
+        sync_deployment_agreements(
+            mpt_client,
+            agreement,
+            customer,
+            adobe_deployments,
+            active_deployment_agreements,
+            adobe_subscriptions,
+            dry_run=dry_run,
+            sync_prices=sync_prices
+        )
+        _process_orphaned_deployment_subscriptions(
```

Comment:
As far as I know, this method should be called for all agreements unless we already identified the use case for this logic. If that's the case, please add a short comment explaining it.

Please, check also the main branch, the logic was changed there

<img width="788" height="305" alt="image" src="https://github.com/user-attachments/assets/69cdd6da-b665-4f00-839b-2faa1ffbaaba" />